src/runtime/update-component.ts

```diff
@@ -266,7 +266,12 @@ const callRender = (hostRef: d.HostRef, instance: any, elm: HTMLElement, isIniti
           renderVdom(hostRef, instance, isInitialLoad);
         }
       } else {
-        elm.textContent = instance;
+        const shadowRoot = elm.shadowRoot;
+        if (BUILD.shadowDom) {
```

Comment:
Should this be checking `hostRef.$cmpMeta$.flags & CMP_FLAGS.shadowDomEncapsulation` instead? I don't think it's guaranteed that this component has a shadow root just because the build flag is set.

Reply:
Good catch @tanner-reits 👌 updated the if statement.